### PR TITLE
cleanup(code): Compilation warnings unused parameters, fallthrough and forbidden uses

### PR DESCRIPTION
**Pull request by sechkem**
During the compilation process, several warnings have been identified that need to be removed for more maintainable code.

REFS: MON-151231

### CODE REVIEW

File: clib/test/main.cc [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 #include <gtest/gtest.h>
 
-int main(int argc, char* argv[], char** env) {
+int main(int argc, char* argv[], char** env [[maybe_unused]]) {
```
Comment (bouda1) [inline clib/test/main.cc:21]: Isn't it possible to simply remove env argument?

File: engine/src/commands/commands.cc [hunk 2]
```diff
@@ -725,28 +725,33 @@ int cmd_process_host_check_result(int cmd, time_t check_time, char* args) {
...
-    ++output;
-  } else
-    output = "";
-  int return_code(strtol(delimiter, nullptr, 0));
+  std::string host_name = std::string(*split_it);
 
-  // replace \\n with \n
-  string::unescape(output);
+  int return_code;
+  ++split_it;
```
Comment (bouda1) [inline engine/src/commands/commands.cc:738]: split_it can be split.end(), isn't it?
So you have to check this

File: engine/src/commands/commands.cc [hunk 3]
```diff
@@ -725,28 +725,33 @@ int cmd_process_host_check_result(int cmd, time_t check_time, char* args) {
...
+    output = split_it->data();
+    // replace \\n with \n
+    string::unescape(output);
+  }
 
   // Submit the check result.
-  return (
-      process_passive_host_check(check_time, host_name, return_code, output));
+  return (process_passive_host_check(check_time, host_name.c_str(), return_code,
+                                     output.c_str()));
```
Comment (bouda1) [inline engine/src/commands/commands.cc:754]:
```suggestion
  return process_passive_host_check(check_time, host_name.c_str(), return_code,
                                     output.c_str());
```

File: clib/test/main.cc (hunk 1 above)
Comment (sechkem) [reply clib/test/main.cc:21]: I will remove

File: engine/src/commands/commands.cc (hunk 2 above)
Comment (sechkem) [reply engine/src/commands/commands.cc:738]: yes,i will fix.

**Review by bouda1 — APPROVED** [review]

File: engine/src/commands/commands.cc [hunk 4]
```diff
@@ -725,28 +725,37 @@ int cmd_process_host_check_result(int cmd, time_t check_time, char* args) {
...
+
+  if (split_it == split.end())
+    return ERROR;
+
+  if (!absl::SimpleAtoi(*split_it, &return_code))
+    return ERROR;
+
+  ++split_it;
+  std::string output;
+  if (split_it == split.end()) {
```
Comment (jean-christophe81) [inline engine/src/commands/commands.cc:748]: replace by ```
if (split_it != split.end() {
   output = split_it->data();
    // replace \\n with \n
    string::unescape(output);
}
```

**Review by jean-christophe81 — APPROVED** [review]

**Pull request merged**